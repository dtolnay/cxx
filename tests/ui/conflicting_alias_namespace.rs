#[cxx::bridge(namespace = correct)]
mod here {
    extern "C" {
        type StringPiece;
    }
}

// Rustfmt mangles the extern type alias.
// https://github.com/rust-lang/rustfmt/issues/4159
#[rustfmt::skip]
#[cxx::bridge(namespace = other)]
#[cxx::alias_namespace(crate::here = correct)]
#[cxx::alias_namespace(crate::here = folly)]
mod there {
    extern "C" {
        type StringPiece = crate::here::StringPiece;
    }
}

fn main() {}
