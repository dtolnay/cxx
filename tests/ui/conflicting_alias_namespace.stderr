error: conflicting cxx::alias_namespace attributes for crate::here: correct, folly
